controller/api/destination/endpoint_translator_test.go

```diff
@@ -0,0 +1,272 @@
+		mockGetServer, translator := makeEndpointTranslator(t)
+
+		translator.Add(mkPodSet(normalPod))
+		translator.Remove(mkPodSet(tlsOptionalPod))
+
+		expectedNumUpdates := 2
+		actualNumUpdates := len(mockGetServer.updatesReceived)
+		if actualNumUpdates != expectedNumUpdates {
+			t.Fatalf("Expecting [%d] updates, got [%d]. Updates: %v", expectedNumUpdates, actualNumUpdates, mockGetServer.updatesReceived)
+		}
+	})
+
+	t.Run("Sends addresses as removed or added", func(t *testing.T) {
+		mockGetServer, translator := makeEndpointTranslator(t)
+
+		translator.Add(mkPodSet(normalPod, tlsOptionalPod))
+		translator.Remove(mkPodSet(tlsDisabledPod))
+
+		addressesAdded := mockGetServer.updatesReceived[0].GetAdd().Addrs
+		actualNumberOfAdded := len(addressesAdded)
+		expectedNumberOfAdded := 2
+		if actualNumberOfAdded != expectedNumberOfAdded {
+			t.Fatalf("Expecting [%d] addresses to be added, got [%d]: %v", expectedNumberOfAdded, actualNumberOfAdded, addressesAdded)
+		}
+
+		addressesRemoved := mockGetServer.updatesReceived[1].GetRemove().Addrs
+		actualNumberOfRemoved := len(addressesRemoved)
+		expectedNumberOfRemoved := 1
+		if actualNumberOfRemoved != expectedNumberOfRemoved {
+			t.Fatalf("Expecting [%d] addresses to be removed, got [%d]: %v", expectedNumberOfRemoved, actualNumberOfRemoved, addressesRemoved)
+		}
+
+		sort.Slice(addressesAdded, func(i, j int) bool {
+			return addressesAdded[i].GetAddr().Port < addressesAdded[j].GetAddr().Port
+		})
+		checkAddressAndWeight(t, addressesAdded[0], normalPod)
+		checkAddressAndWeight(t, addressesAdded[1], tlsOptionalPod)
+		checkAddress(t, addressesRemoved[0], tlsDisabledPod)
+	})
+
+	t.Run("Sends metric labels with added addresses", func(t *testing.T) {
+		mockGetServer, translator := makeEndpointTranslator(t)
+
+		translator.Add(mkPodSet(normalPod))
+
+		actualGlobalMetricLabels := mockGetServer.updatesReceived[0].GetAdd().MetricLabels
+		expectedGlobalMetricLabels := map[string]string{"namespace": "service-ns", "service": "service-name"}
+		if !reflect.DeepEqual(actualGlobalMetricLabels, expectedGlobalMetricLabels) {
+			t.Fatalf("Expected global metric labels sent to be [%v] but was [%v]", expectedGlobalMetricLabels, actualGlobalMetricLabels)
+		}
+
+		actualAddedAddress1MetricLabels := mockGetServer.updatesReceived[0].GetAdd().Addrs[0].MetricLabels
+		expectedAddedAddress1MetricLabels := map[string]string{
+			"pod":                   "pod1",
+			"replicationcontroller": "rc-name",
+			"serviceaccount":        "serviceaccount-name",
+			"control_plane_ns":      "linkerd",
+		}
+		if !reflect.DeepEqual(actualAddedAddress1MetricLabels, expectedAddedAddress1MetricLabels) {
+			t.Fatalf("Expected global metric labels sent to be [%v] but was [%v]", expectedAddedAddress1MetricLabels, actualAddedAddress1MetricLabels)
+		}
+	})
+
+	t.Run("Sends TlsIdentity when enabled", func(t *testing.T) {
+		expectedTLSIdentity := &pb.TlsIdentity_DnsLikeIdentity{
```

Comment:
TIOLI, I'm curious about the different capitalization of "TlsIdentity" vs "TLSIdentity" which is a little confusing when reading through. Is this intentional?

Reply:
I believe that go style is to capitalize the entire acronym (like `actualTLSIdentity`) in identifiers, unless they are at the start of of a private identifier (like `tlsDisablePod`).  However, the `TlsIdentity` field and related methods are code-generated from protobuf and therefore don't follow this language specific convention.